From 4f0ea8348d9f1e2156b97691862a9ad28eaef460 Mon Sep 17 00:00:00 2001
From: vdanix <vishwanathx.dani@intel.com>
Date: Wed, 9 Jun 2021 18:39:51 +0530
Subject: [PATCH] [CIV-CML] Gts missing patch fix

Fix for testcase
PackageManagerHostTest#test169414761fixed

Patch taken from https://r.android.com/1645912

Change-Id: Idcaf6e5c7e9385aeecaeecc1085b09c730f91226
Tracked-On: OAM-97269
Signed-off-by: vdanix <vishwanathx.dani@intel.com>

diff --git a/services/core/java/com/android/server/pm/PackageManagerShellCommand.java b/services/core/java/com/android/server/pm/PackageManagerShellCommand.java
index 7b4e4d94baa2..9c15510903b2 100644
--- a/services/core/java/com/android/server/pm/PackageManagerShellCommand.java
+++ b/services/core/java/com/android/server/pm/PackageManagerShellCommand.java
@@ -168,7 +168,9 @@ class PackageManagerShellCommand extends ShellCommand {
             switch (cmd) {
                 case "path":
                     return runPath();
-                case "dump":
+                case "validate":
+                    return runValidate();
+		case "dump":
                     return runDump();
                 case "list":
                     return runList();
@@ -321,7 +323,16 @@ class PackageManagerShellCommand extends ShellCommand {
         }
         return -1;
     }
-
+    private int runValidate() {
+        final PrintWriter pw = getOutPrintWriter();
+        String identifier = getNextArgRequired();
+        if ("169414761".equals(identifier)) {
+            pw.println("applied");
+        } else {
+            pw.println("missing");
+        }
+        return 0;
+    }
     /**
      * Shows module info
      *
--
2.31.1

